### PR TITLE
update README

### PR DESCRIPTION
**Pull request by mohsinulhaq**
https://github.com/yannickcr/eslint-plugin-react/issues/1955#issuecomment-451182940

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -39,7 +39,7 @@ You should also specify settings that will be shared across all the plugin rules
       "createClass": "createReactClass", // Regex for Component Factory to use,
                                          // default to "createReactClass"
       "pragma": "React",  // Pragma to use, default to "React"
-      "version": "15.0", // React version, default to the latest React stable release
+      "version": "detect", // React version, "detect" automatically picks the version from your project's package.json. "detect" will be the default value in future versions
```
Comment (ljharb) [inline README.md]: Let's make sure to document both "detect" and the explicit form, like so:
```suggestion
      "version": "detect", // React version, "detect" automatically picks the version from your project's package.json. You can also use `16.0`, `16.3`, etc, if you want to override the detection value.
```

**Comment by mohsinulhaq** [comment]
is it ok now?

File: README.md [hunk 2]
```diff
@@ -39,7 +39,9 @@ You should also specify settings that will be shared across all the plugin rules
       "createClass": "createReactClass", // Regex for Component Factory to use,
                                          // default to "createReactClass"
       "pragma": "React",  // Pragma to use, default to "React"
-      "version": "15.0", // React version, default to the latest React stable release
+      "version": "detect", // React version, "detect" automatically picks the version from your project's package.json.
```
Comment (alexzherdev) [inline README.md]: This is technically not entirely correct though. `package.json` (generally) defines a range, whereas "detect" will detect the exact version you have installed.

Comment (mohsinulhaq) [reply README.md]: yes, you are right, so do we actually check package lock?

Comment (alexzherdev) [reply README.md]: I'll defer the final wording to @ljharb though.

Comment (mohsinulhaq) [reply README.md]: okay that makes sense.

**Comment by mohsinulhaq** [comment]
Thanks! This was my first OSS PR to a well-known library. 😃

**Pull request merged**